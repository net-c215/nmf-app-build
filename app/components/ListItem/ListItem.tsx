import React from "react";
import { TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import styles from "./ListItem.styles";
import Text from "../Text";
import { Colors } from "../../style";

interface Props {
  title: string;
  onPress: () => void;
}

const ListItem = ({ title, onPress }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text.Primary numberOfLines={1} bold style={styles.text}>
        {title}
      </Text.Primary>
      <Ionicons
        style={styles.icon}
        name={"ios-arrow-forward"}
        size={26}
        color={Colors.grey40}
      />
    </TouchableOpacity>
  );
};

export default ListItem;
