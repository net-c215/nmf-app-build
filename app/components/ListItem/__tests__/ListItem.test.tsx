import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import ListItem from "../ListItem";

describe("<ListItem />", () => {
  const defaultProps = {
    selected: true,
    title: "France",
    onPress: () => {
      // do nothing.
    },
  };
  const wrapper = renderer.create(<ListItem {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
