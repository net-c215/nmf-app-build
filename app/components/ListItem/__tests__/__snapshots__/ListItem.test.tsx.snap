// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> render 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text.Primary
    bold={true}
    numberOfLines={1}
    style={
      Object {
        "paddingVertical": 20,
      }
    }
  >
    France
  </Text.Primary>
  <Ionicons
    color="#9CA6A0"
    name="ios-arrow-forward"
    size={26}
    style={
      Object {
        "marginTop": 18,
      }
    }
  />
</TouchableOpacity>
`;
